ocaml/xapi/xapi_pool_update.ml

```diff
@@ -150,7 +150,12 @@ let with_api_errors f x =
   | Smint.Command_killed(ret, status, stdout_log, stderr_log) ->
     let msg = Printf.sprintf "Smint.Command_{failed,killed} ret = %d; status = %s; stdout = %s; stderr = %s"
         ret status stdout_log stderr_log in
-    raise (Api_errors.Server_error (Api_errors.internal_error, [msg]))
+    error "%s" msg;
+    raise (Api_errors.Server_error (Api_errors.invalid_update, ["Please upload the valid package."]))
```

Comment:
I think the message should read either `Please upload a valid package` or `The uploaded package is invalid`. There is not enough context to talk about "the valid package". Since the message is used twice, you might want to let-bind it.